lib/std/json.zig

```diff
@@ -1530,7 +1532,26 @@ test "skipValue" {
     }
 }
 
-fn parseInternal(comptime T: type, token: Token, tokens: *TokenStream, options: ParseOptions) !T {
+const ParseInternalError = error{
+    AllocatorRequired,
+    DuplicateJSONField,
+    InvalidEnumTag,
+    InvalidNumber,
+    MissingField,
+    NoUnionMembersMatched,
+    Overflow,
+    UnexpectedEndOfJson,
+    UnexpectedToken,
+    UnexpectedValue,
+    UnknownField,
+} || SkipValueError || TokenStream.Error || error{OutOfMemory} || std.fmt.ParseIntError;
```

Comment:
`ParseInternalError` intentionally doesn't have a specified error set, as different errors are only possible/reachable with certain types passed.
e.g. `OutOfMemory` cannot be hit when `T` isn't a pointer type.

Reply:
Okay, I tried to do your suggested approach. Now I don't know how to avoid infinite recursion. In the test I wrote it goes like this:
1. Check union errors, e.g. check all union fields
2. i64 - no errors, next
3. Array - check child field type
4. Child field type is union, check all union fields
5. ...

I can't check whether it is going to be a recursion because I need to retrieve a parent of my parent, the sequence is Union -> Array -> Union. And of course more complex scenarios are possible.

Reply:
there lies the rub.

That's why the compiler couldn't figure it out either :P

The json parser accepting pointer types opens it up for infinite recursion; and currently this can lead to unbounded stack usage! Given the future goals of zig include limiting unbounded stack usage, perhaps the solution is to remove pointers to complex objects from the accepted input of `json.parse`.

Reply:
Sorry, I don't think I understand your suggestion. Do you suggest to raise a compile error whenever there's a recursive type definition we are trying to parse? This is the exact reason for this PR, I want to be able to parse into recursive definitions. Maybe I misunderstand something.

Also, can I do it like this? commit `Don't analyze already inferred types`

Reply:
By the way, maybe I wasn't clear enough. By "infinite recursion" I meant that compiler produces an error that eval quota is overreached (`@setEvalBranchQuota`), it already saves from infinite recursion scenario.